app/javascript/src/LoginPage/utils/formValidation.js

```diff
@@ -0,0 +1,68 @@
+// @flow
+
+import validate from 'validate.js'
+
+const constraintsTypes = {
+  text: {
+    presence: true,
+    length: {minimum: 1}
+  },
+  email: {
+    presence: true,
+    email: true,
+    length: {minimum: 1}
+  },
+  password: {
+    presence: true,
+    length: {minimum: 6}
+  },
+  password_confirmation: {
+    presence: true,
+    length: {minimum: 1},
+    equality: {
+      attribute: 'user[password]'
+    }
+  }
+}
+
+const constraints = {
+  'new_session': {
+    'username': constraintsTypes.text,
+    'password': constraintsTypes.password
+  },
+  'request_password': {
+    'email': constraintsTypes.email
+  },
+  'signup_form': {
+    'user[username]': constraintsTypes.text,
+    'user[email]': constraintsTypes.email,
+    'user[password]': constraintsTypes.password,
+    'user[password_confirmation]': constraintsTypes.password_confirmation
+  }
+}
+
+const validateAllFields = (form: HTMLFormElement) => {
+  const errors = validate(form, constraints[form.id])
+
+  if (errors) {
```

Comment:
So we are using validate.js to validate the form, but then we are overriding its format with `false` ?
Isn't there a way to just tell validate to return that already?
if not, still having doubts, but you could simplify this with...

```
return !!errors ? Object.keys(errors).reduce((object, item) => { obj[item] = false; return obj }, {}) : {}
```
10 lines in 1 ;) 

PS: I'm still not sure if using validate.js and override its format...

Reply:
I can try passing a custom validator